tests/set_test.go

```diff
@@ -115,6 +115,11 @@ func TestSetWithOptions(t *testing.T) {
 			commands: []string{"SET k v XX EX 1", "GET k", "SLEEP 2", "GET k", "SET k v XX EX 1", "GET k"},
 			expected: []interface{}{"(nil)", "(nil)", "OK", "(nil)", "(nil)", "(nil)"},
 		},
+		{
```

Comment:
Please add tests for error conditions as well.

Reply:
Sure, I'll add the error test cases too 👍